rclrs/src/context.rs

```diff
@@ -3,73 +3,92 @@ use crate::rcl_bindings::*;
 use crate::Node;
 use alloc::sync::Arc;
 use alloc::vec::Vec;
-use cstr_core::{c_char, CString};
+use std::string::String;
 
 #[cfg(not(feature = "std"))]
-use spin::{Mutex, MutexGuard};
+use cstr_core::{c_char, CString};
+#[cfg(not(feature = "std"))]
+use spin::Mutex;
 
 #[cfg(feature = "std")]
-use parking_lot::{Mutex, MutexGuard};
-
-pub(crate) struct ContextHandle(Mutex<rcl_context_t>);
-
-impl ContextHandle {
-    pub fn get_mut(&mut self) -> &mut rcl_context_t {
-        self.0.get_mut()
-    }
-
-    pub fn lock(&self) -> MutexGuard<rcl_context_t> {
-        self.0.lock()
-    }
-}
+use cty::c_char;
+#[cfg(feature = "std")]
+use parking_lot::Mutex;
+#[cfg(feature = "std")]
+use std::ffi::CString;
 
-impl Drop for ContextHandle {
+impl Drop for rcl_context_t {
```

Comment:
This change is meant to start a discussion, I'm not dead set on it. But a priori it makes sense to me to implement Drop directly on the rcl structs and not to wrap them if they're not going to be exposed to users anyway.

Reply:
I'm a bit decisive about it, on the one hand, I'd rather not expose the `rcl` API directly to Rust, but also it makes no sense to keep `ContextHandle` if users can't access it and we can implement traits directly. I'd go with removing `ContextHandle` and bring it back if needed.